### PR TITLE
Unify the 3 TokenCountEstimators into 2.

### PR DESCRIPTION
**Pull request by crutcher**

### CODE REVIEW

File: langchain4j-core/src/main/java/dev/langchain4j/model/chat/ChatTokenCountEstimator.java [hunk 1]
```diff
@@ -14,7 +13,7 @@
  * Represents an interface for estimating the count of tokens in various text types such as a text, message, prompt, text segment, etc.
  * This can be useful when it's necessary to know in advance the cost of processing a specified text by the LLM.
  */
-public interface TokenCountEstimator {
+public interface ChatTokenCountEstimator extends TokenCountEstimator {
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/model/chat/ChatTokenCountEstimator.java:16]: Is this renaming required? `ChatTokenCountEstimator` is in the `chat` package, so there is a distinction

File: langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java [hunk 2]
```diff
@@ -198,7 +198,7 @@ private Response<AiMessage> generate(List<ChatMessage> messages,
     }
 
     @Override
-    public int estimateTokenCount(List<ChatMessage> messages) {
+    public int estimateChatMessagesTokenCount(List<ChatMessage> messages) {
```
Comment (langchain4j) [inline langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java:201]: I would avoid making breaking changes when not necessary

**Review by ghost — COMMENTED** [review]
@crutcher thank you! Could you pelase provide a motivation behing this PR?

I was planning to revisit the design of these estimators and `Tokenizer` soon, and anticipate some breaking changes. Since this PR is not urgent, I would propose to put it on hold for a bit, to avoid making 2 breaking changes in a row. Hope for your understanding!

File: langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java (hunk 2 above)
Comment (crutcher) [reply langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java:201]: if we unify, we need to make sure that the interfaces are type erasure compatible.

we can't have multiple methods with the same erasure, so these became:
`estimateTokenCount(List<T> ...)` and caused conflicts.

File: langchain4j-core/src/main/java/dev/langchain4j/model/chat/ChatTokenCountEstimator.java (hunk 1 above)
Comment (crutcher) [reply langchain4j-core/src/main/java/dev/langchain4j/model/chat/ChatTokenCountEstimator.java:16]: I see that argument; but it does really impact IDE code completion

File: langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java (hunk 2 above)
Comment (langchain4j) [reply langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java:201]: Sure, but we can keep interfaces separately and there will be no problem with type erasure. As well as no breaking changes for users

**Pull request closed without merging**